File: go/wasm/meta.go
// +build js,wasm

package main

import (
	"github.com/cosmos/amino-js/go/src"
	"syscall/js"
)

func DecodeDisambPrefixBytes(this js.Value, args []js.Value) interface{} {
	db, hasDb, pb, hasPb, n, err := src.DecodeDisambPrefixBytes(typedArrayToByteSlice(args[0]))
	return []interface{}{js.TypedArrayOf(db), hasDb, js.TypedArrayOf(pb), hasPb, n, errorOf(err)}
}

func NameToDisfix(this js.Value, args []js.Value) interface{} {
	pb, db := src.NameToDisfix(args[0].String())
	return []interface{}{js.TypedArrayOf(pb), js.TypedArrayOf(db)}
}

func ByteSliceSize(this js.Value, args []js.Value) interface{} {
	bz := typedArrayToByteSlice(args[0])
	return src.ByteSliceSize(bz)
}

func UvarintSize(this js.Value, args []js.Value) interface{} {
	return src.UvarintSize(uint64(args[0].Int()))
}

func VarintSize(this js.Value, args []js.Value) interface{} {
	return src.VarintSize(int64(args[0].Int()))
}

File: src/wasm/meta.ts
import * as Amino from '../../lib/Amino.wasm';
import { AminoBytes, DisambBytes, PrefixBytes } from '../../lib/types';

export {
    nameToDisfix,
    byteSliceSize,
    uvarintSize,
    varintSize
} from '../../lib/Amino.wasm';

export function decodeDisambPrefixBytes (amino: AminoBytes): [DisambBytes | null, PrefixBytes | null, number] {
    const [disamb, hasDisamb, prefix, hasPrefix, length, error] = Amino.decodeDisambPrefixBytes(amino);
    if (error) {
        throw new Error(error);
    }
    return [
        hasDisamb ? disamb : null,
        hasPrefix ? prefix : null,
        length
    ];
}
